lib/ioki/model/base.rb

```diff
@@ -174,6 +177,17 @@ def constantize_in_module(class_name)
 
         self.class.module_parent.const_get class_name
       end
+
+      def class_name_from_value_type(class_name, value)
+        return class_name unless class_name.is_a?(Array)
+        return nil if value.nil?
+
+        class_name_from_value_type = value['type'].split('_').map(&:capitalize).join('')
```

Comment:
I like this extension.. but i keep thinking about whether we need more "generic" specs for classes using these features...

Like a spec that uses a dummy class inheriting from model base and then using an array in this case here.

Reply:
Yes! That sounds like a good idea!

I was wondering how to add proper tests for this extension because the existing spec is running with vcr recorded examples I think. 🤔 

But I like the dummy class and inheriting from the base class idea more tbh. I will give it a try!

Reply:
**UPDATE**
I added some specs for the TaskList Model using a dummy class inheriting from the Ioki::Model::Base class.